source/common/upstream/upstream_impl.h

```diff
@@ -412,6 +412,15 @@ class ClusterImplBase : public Cluster, protected Logger::Loggable<Logger::Id::u
    */
   void setOutlierDetector(const Outlier::DetectorSharedPtr& outlier_detector);
 
+  /**
+   * Wrapper around Network::Address::resolveProtoAddress() to catch the specific error in relation
+   * with envoy::api::v2::Cluster::DiscoveryType hence we can throw more meaningful message.
+   * @param address supplies the address proto to resolve.
+   * @return pointer to the Instance from a envoy::api::v2::core::Address.
```

Comment:
Our convention is to list the return type first. So something like:

`@return Network::Address::InstanceConstSharedPtr the resolved address.`

Reply:
Noted. Thanks! Changed here cf5a97b